common/djangoapps/student/views.py

```diff
@@ -1816,7 +1820,7 @@ def auto_auth(request):
 
     # Enroll the user in a course
     if course_key is not None:
-        CourseEnrollment.enroll(user, course_key)
+        CourseEnrollment.enroll(user, course_key, mode=enrollment_mode if enrollment_mode else 'honor')
```

Comment:
nitpick: it might be a bit more cleaner to make the 'honor' the default on the request.GET.get('enrollment_mode', 'honor')